x/stake/client/rest/tx.go

```diff
@@ -86,182 +74,92 @@ func delegationsRequestHandlerFn(cdc *codec.Codec, kb keys.Keybase, cliCtx conte
 			return
 		}
 
```

Comment:
I think we also need to support simulation and generation only?

Reply:
This needs to be added in all the endpoints

Reply:
While this may be true, the lines proceeding this will still require a password and chain ID:

```
baseReq := req.BaseReq.Sanitize()
 		if !baseReq.ValidateBasic(w) {
 			return
 		}
```

Reply:
I'm not following... those are checked in the `ValidateBasic` function

Reply:
We can't have `generate_only` and `ValidateBasic` together (e.g. we don't need the password and maybe even chain ID). So `ValidateBasic` should take a `genOnly` boolean or something.

Reply:
added